cli/compose/loader/loader.go

```diff
@@ -352,20 +357,34 @@ func resolveEnvironment(serviceConfig *types.ServiceConfig, workingDir string) e
 		}
 
 		for k, v := range runconfigopts.ConvertKVStringsToMap(envVars) {
-			environment[k] = v
+			interpolatedV, ok := lookupEnv(k)
+			if ok {
+				// lookupEnv is prioritized over the file
+				environment[k] = interpolatedV
+			} else {
+				environment[k] = v
+			}
```

Comment:
Here's the logic from Compose: https://github.com/docker/compose/blob/master/compose/config/config.py#L589-L598

I think we can reduce the size of this function quite a bit by matching the logic.

Once `delete()` is removed from below, i think these blocks will be nearly identical. So we could have a function like this:

```go

func updateEnvironment(environment, map[string]string, vars map[string]string, lookupEnv template.Mapping) map[string[string {
    ...
}

environment = updateEnvironment(environment, runconfigopts.ConvertKVStringsToMap(envVars), lookupEnv)
serviceConfig.Environment = updateEnvironment(environment, serviceConfig.Environment, lookupEnv)
return nil

```